### DISCUSSION TITLE
is there a way to override the default prompt in the assistant panel?

### DISCUSSION THREAD

**Discussion by kmrgirish** [zed-industries/zed#21053] [Feature Requests]
I would like to change the default prompt to add some contextual information, e.g. always output code in golang.

<img width="369" alt="image" src="https://github.com/user-attachments/assets/92e3491d-b145-49e7-918b-c5790913a09e">

**Comment by SomeoneToIgnore** [comment]
Should be possible if you open the prompt library:

![image](https://github.com/user-attachments/assets/74b77738-4446-473b-a663-e47f6c28d21f)

create/edit prompts and click those star-like icons to determine which of the library prompts come as defaults:

![image](https://github.com/user-attachments/assets/a31dd601-439f-42a3-ba63-563edfc7dc21)

**Reply by 0xgirish** [reply]
Thank you for the reply—it worked perfectly!

I was wondering if it’s possible to configure default prompts on a per-project basis. For example, if I am working on Project A, Default Prompt X should automatically be used, and for Project B, Default Prompt Y should be applied. This would allow me to provide project-specific contextual information within the default prompt for better efficiency.

Is such a feature available or planned?

**Reply by SomeoneToIgnore** [reply]
I do not think anything LLM-related is available in project-local configs and not sure about the plans.
cc @rtfeldman as the one who considered something related in the past.
